### DISCUSSION TITLE
Upload-artifact : Variable name

### DISCUSSION THREAD

**Discussion by mciolkosz** [community/community#26012] [Actions]
<p>Hi All,</p>
<p>I’ve been looking for ways to have a varying name for the artifact that I upload, but I haven’t found solutions on here that allow what I’m looking for. For reference, I’ve looked at both of these solutions:</p>
<ul>
<li><a href="https://github.community/t/github-actions-using-non-fixed-names-in-upload-artifact/17353/12">https://github.community/t/github-actions-using-non-fixed-names-in-upload-artifact/17353/12</a></li>
<li><a href="https://github.community/t/use-variables-in-upload-artifact/16577">https://github.community/t/use-variables-in-upload-artifact/16577</a></li>
</ul>
<p>What I’m looking for is a way to set the name based on items that are set in a file that is also in the repository. Specifically, I’m using the Gradle for Java Action, with some minor tweaks. I am specifically looking for my artifact to be named <code>&lt;base_name&gt;-&lt;game_version&gt;-&lt;project_version&gt;.jar</code>, where my <code>gradle.properties</code> file in the repo contains</p>
<pre><code class="lang-auto">base_name = exampleproject
game_version = 2.4.3
project-version = 0.3.2-SNAPSHOT
</code></pre>
<p>Thus the output would be <code>exampleproject-2.4.3-0.3.2-SNAPSHOT.jar</code></p>
<p>If this is possible, please let me know how to specify that in my .yml file. If it can’t happen, then I’ll have to figure something else out.</p>
<p>Thanks</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You’ll have to do this in two steps: Assemble the artifact name (or components) from the file and store it in a variable, and then use that in the <code>upload-artifact</code> step. I’d do the first part with a few lines of Python:</p>
<pre><code class="lang-yaml">- run: |
    prop = dict()
    with open('gradle.properties') as fh:
        for line in fh:
            name, value = line.split('=', 1)
            prop[name.strip()] = value.strip()

    print(f'::set-env name=artifact_name::{prop["base_name"]}-'
          f'{prop["game_version"]}-{prop["project-version"]}.jar')
  shell: python
- uses: actions/upload-artifact@v2
  with:
    name: ${{ env.artifact_name }}
    path: # your file goes here
</code></pre>
